internal/v1/server.go

```diff
@@ -140,12 +144,40 @@ func Attach(conf *ServerConfig) (*Server, error) {
 		prometheus.StatusMiddleware,
 	}
 
+	// Global middleware: extract identity early
+	if conf.FedoraAuth {
+		s.echo.Pre(echo.WrapMiddleware(fedora_identity.Extractor))
+	} else {
+		s.echo.Pre(echo.WrapMiddleware(identity.Extractor))
+	}
+
+	// Global middleware: extract or generate trace information
+	s.echo.Pre(strc.NewEchoV4MiddlewareWithConfig(slog.Default(), strc.MiddlewareConfig{
+		Filters: []strc.Filter{
+			strc.IgnorePathPrefix("/metrics"),
+			strc.IgnorePathPrefix("/status"),
+			strc.IgnorePathPrefix("/ready"),
+		},
+	}))
+
+	// Global middleware: add context to echo logger proxy
+	s.echo.Use(func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			c.SetLogger(echoproxy.NewProxyWithContextFor(slog.Default(), c.Request().Context()))
+			return next(c)
+		}
+	})
+
+	// Global middleware: log stack traces into standard logger as error (instead of stdout)
+	s.echo.Use(middleware.RecoverWithConfig(middleware.RecoverConfig{
+		LogLevel: glog.ERROR,
+	}))
+
+	// Route-based middleware:
 	var middlewares []echo.MiddlewareFunc
 
-	if s.fedoraAuth {
-		middlewares = append(middlewaresNoAuth, echo.WrapMiddleware(fedora_identity.Extractor))
-	} else {
-		middlewares = append(middlewaresNoAuth, echo.WrapMiddleware(identity.Extractor), echo.WrapMiddleware(identity.BasePolicy))
+	if !s.fedoraAuth {
+		middlewares = append(middlewaresNoAuth, echo.WrapMiddleware(identity.BasePolicy))
 	}
```

Comment:
Extractors were moved up in this file and added as `Pre` middleware - they must run in the front of the middleware chain so context has necessary values early.